packages/react-form-state/src/FormState.ts

```diff
@@ -212,21 +207,31 @@ export default class FormState<
     } else {
       this.setState({submitting: false, errors});
     }
-  }
+  };
 
-  @memoize()
-  @bind()
-  private fieldWithHandlers<Key extends keyof Fields>(
+  private fieldWithHandlers = <Key extends keyof Fields>(
     field: FieldStates<Fields>[Key],
-    fieldPath: Key,
-  ) {
-    return {
-      ...(field as FieldState<Fields[Key]>),
-      name: fieldPath,
-      onChange: this.updateField.bind(this, fieldPath),
-      onBlur: this.blurField.bind(this, fieldPath),
+    fieldPath: string & Key,
+  ) => {
+    const hashKey = `${fieldPath}:${JSON.stringify(field)}`;
```

Comment:
memoize can be replaced by storing a private cache for return values, in the form of a map, and caching based on a computed hash key.

Reply:
Is the stringify here a bit too much? Is there another way we can compute this hash key?

Reply:
Code before likely used a `WeakMap`, now this is almost certainly leaky.

Reply:
Hmm actually maybe not if this is an instance member on the class...

Reply:
Ooh, interesting...

Reply:
I don't think `WeakMap` would work here, TBH. We need to cache handlers based only on the field path, but still ensure that pure components receive the exact same object reference unless the field data changes. I'll think about this a bit more and try to find a more efficient solution. @TheMallen, what should we do in a case where the field state is `deepEqual`, but the reference is different? Should we just pass through the object? OR should we return a cached reference as per the test suite (wrt. re-rendering PureComponents)?

Reply:
- we don't need to cache them based on field path (the old memoize was based on the `field` object itself as it was the first param)
- don't worry about deep equality, we control the way this stuff is updated and it's all done immutably